client/state/posts/reducer.js

```diff
@@ -163,40 +170,33 @@ export const queries = ( () => {
 
 	return createReducer( {}, {
 		[ POST_RESTORE ]: ( state, { siteId, postId } ) => {
-			return applyToManager( state, siteId, 'receive', {
+			return applyToManager( state, siteId, 'receive', false, {
 				ID: postId,
 				status: '__RESTORE_PENDING'
 			}, { patch: true } );
 		},
 		[ POST_RESTORE_FAILURE ]: ( state, { siteId, postId } ) => {
-			return applyToManager( state, siteId, 'receive', {
+			return applyToManager( state, siteId, 'receive', false, {
 				ID: postId,
 				status: 'trash'
 			}, { patch: true } );
 		},
 		[ POSTS_REQUEST_SUCCESS ]: ( state, { siteId, query, posts, found } ) => {
-			if ( ! state[ siteId ] ) {
-				state = {
-					...state,
-					[ siteId ]: new PostQueryManager()
-				};
-			}
-
-			return applyToManager( state, siteId, 'receive', posts, { query, found } );
+			return applyToManager( state, siteId, 'receive', true, posts, { query, found } );
 		},
 		[ POSTS_RECEIVE ]: ( state, { posts } ) => {
 			return reduce( groupBy( posts, 'site_ID' ), ( memo, sitePosts, siteId ) => {
-				return applyToManager( memo, siteId, 'receive', sitePosts );
+				return applyToManager( memo, siteId, 'receive', true, sitePosts );
```

Comment:
Nice thing about this is that theoretically if we wanted to add "guessing" behavior to `QueryManager` (based on query logic), we could show posts that had been received but not necessarily associated with a particular query, even before a request for that query has completed. For example, loading the editor for a published post and then navigating to published posts, we know that the post we'd just been editing should be included in that list, so we could show it immediately.